src/main/java/org/trigon/hardware/misc/servo/ServoIO.java

```diff
@@ -0,0 +1,44 @@
+package org.trigon.hardware.misc.servo;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import org.littletonrobotics.junction.AutoLog;
+import org.trigon.hardware.RobotHardwareStats;
+import org.trigon.hardware.misc.servo.io.RealServoIO;
+import org.trigon.hardware.misc.servo.io.SimulationServoIO;
+
+import java.util.function.DoubleSupplier;
+
+public class ServoIO {
+    static ServoIO generateServoIO(int channel) {
+        if (RobotHardwareStats.isReplay())
+            return new ServoIO();
+        if (RobotHardwareStats.isSimulation())
+            return new SimulationServoIO();
+        return new RealServoIO(channel);
+    }
+
+    protected void updateInputs(ServoInputsAutoLogged inputs) {
+    }
+
+    protected void setSpeed(double speed) {
+    }
+
+    protected void setAngle(Rotation2d angle) {
+    }
+
+    protected void setPosition(double position) {
+    }
+
+    protected void setBoundsMicroseconds(int maxPulseWidthMicroseconds, int maxDeadbandRangeMicroseconds, int centerPulseMicroseconds, int minDeadbandRangeMicroseconds, int minPulseWidthMicroseconds) {
+    }
+
+    protected void setSimulationSupplier(DoubleSupplier simulationValueSupplier) {
+    }
```

Comment:
Create an actual viable sim constructor 
Make the target angle logic actually work